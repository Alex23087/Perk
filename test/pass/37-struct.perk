import "stdio.h"

struct Point {
    let x := 10,
    let y := 20,
    let z := 30
}

fun main (): void {
    let p := make Point(y = 42, x = 11);
    printf("%d\n", p.x); // Should print 11
    printf("%d\n", p.y); // Should print 42
    printf("%d\n", p.z); // Should print 30

    let p2 := make Point();
    p = p2;
    p2.x = 100;
    p.x = 200;
    printf("%d\n", p.x); // Should print 200
    printf("%d\n", p2.x); // Should print 100
    printf("%d\n", p2.y); // Should print 20
    printf("%d\n", p.z); // Should print 30
}