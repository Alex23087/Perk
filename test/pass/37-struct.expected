11
42
30
200
100
20
30
